sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/job_limits.py

```diff
@@ -43,6 +44,8 @@ def __init__(self, *, timeout: Optional[int] = None):
         self.timeout = timeout
 
     def _to_rest_object(self) -> RestCommandJobLimits:
+        if isinstance(self.timeout, PipelineInput):
+            return RestCommandJobLimits(timeout=self.timeout)
         return RestCommandJobLimits(timeout=to_iso_duration_format(self.timeout))
 
     @classmethod
```

Comment:
I updated the code but not in this way. For this code there is still the problem which I described in clement's comment: Response dict key for type is "job_limits_type" not "type", so this code will raise error:
`CommandJobLimits.__init__() got an unexpected keyword argument 'job_limits_type'`

And I feel it not approriate if I do the hard code mapping inside the function, because that key "job_limits_type" may change.